server.js

```diff
@@ -1,35 +1,81 @@
 import express from 'express'
-import bodyParser from 'body-parser'
+import listEndpoints from 'express-list-endpoints'
 import cors from 'cors'
+import dotenv from 'dotenv'
 import mongoose from 'mongoose'
+import booksdata from './data/books.json'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
+dotenv.config()
 
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost:27017/project-mongo"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
+const bookSchema = new mongoose.Schema({
+  bookID: Number,
+  title: String,
+  authors: String,
+  average_rating: Number,
+  num_pages: Number
+})
+
+const Book = mongoose.model('Book', bookSchema)
+
+if (process.env.RESET_DB) {
+  const seedDataBase = async () => {
+    await Book.deleteMany()
+
+    await booksdata.forEach((item) => {
+      const newBook = new Book({
+        bookID: item.bookID,
+        title: item.title,
+        average_rating: item.average_rating,
+        num_pages: item.num_pages
+      })
+      newBook.save()
```

Comment:
As I’ve understood it, the await before the foreach loop is not necessary to have here since the foreach is async by itself. My suggestion is to write it instead like this: 

```
    booksdata.forEach(async (item) => {
      const newBook = new Book({
        bookID: item.bookID,
        title: item.title,
        average_rating: item.average_rating,
        num_pages: item.num_pages
      })
      await newBook.save()
    })
```

Reply:
Siri is using awaiting instead of declaring a separate async function inside the forEach 😸